### DISCUSSION TITLE
[question] File list

### DISCUSSION THREAD

**Discussion by dosch** [mkdocs/mkdocs#2338] [Q&A]
Can MKDocs generate a page from a directory with all sorts of files in it?

I have a directory where I save all sorts of file for people to download: different svgs, zips, tars, jpgs etc. etc. Often with quick changing version numbers. 

Maintaining a table with correct links to every file is rather tedious... So I would like to link to the directory where people can use the naming of the respective files to figure out which one they need and download them

Thnx!

**Comment by waylan** [comment]
MkDocs will copy any files in your docs_dir to the site. However, MkDocs will not do anything else with non-markdown files. If you would like to auto-generate a list of files, then you would need to use a plugin for that.

**Comment by fu-sen** [comment]
Are you looking for something like Apache's `Options +Indexes` ? or 
https://vercel.com/docs/edge-network/directory-listing

I know someone who blogs on MkDocs. 
He seems to have manually generated a list of articles in `index.md`.
This idea can also be a means of auto-generation.

Ideally, it will be produced as a plug-in.

**Comment by fu-sen** [comment]
Added to Wiki. Expect the development of plugins:
https://github.com/mkdocs/mkdocs/wiki/Plugins-Ideas

**Comment by fu-sen** [comment]
Originally this post was on an issue, but it has been moved to the recently set up Discussion:
https://github.com/mkdocs/mkdocs/issues/2322

Please continue to comment here if necessary.

... and I'm a MkDocs User, but I've included this comment. 💻😆

**Comment by Andre601** [comment]
Maybe awesome-pages is what you want?

**Comment by fu-sen** [comment]
I think this will change the way MkDocs is used. This can compete with other static site generators.

**Comment by fu-sen** [comment]
If it is possible to display images, for example, it will be possible to display a list of themes. Instead of here:
https://github.com/mkdocs/mkdocs/wiki/MkDocs-Themes

You need to consider how to post the theme.

but anyway, you will be able to use MkDocs for quite a variety of purposes.

**Comment by oprypin** [comment]
https://github.com/oprypin/mkdocs-gen-files plugin can do this

Untested code, to be put in a script as per the plugin:
```python
import os
import mkdocs_gen_files

with mkdocs_gen_files.open('some_dir/files/index.md', 'w') as f:
  for filename in os.listdir('docs/some_dir/files'):
    f.write(f'* [{filename}]({filename})')

**Comment by stratself** [comment]
Shameless plugin: I wrote https://github.com/skedastically/mkdocs-macro-gimmicks/ which can `{{ listfiles() }}` in a page's current directory
